packages/gateway/src/features/user/root-resolvers.ts

```diff
@@ -309,6 +310,44 @@ export const resolvers: GQLResolver = {
       }
       return true
     },
+    async changePhone(
+      _,
+      { userId, phoneNumber, nonce, verifyCode },
+      authHeader
+    ) {
+      if (!isTokenOwner(authHeader, userId)) {
+        return await Promise.reject(
+          new Error(
+            `Change phone is not allowed. ${userId} is not the owner of the token`
+          )
+        )
+      }
+      try {
+        await checkVerificationCode(nonce, verifyCode)
+      } catch (err) {
+        logger.error(err)
+        return await Promise.reject(
+          new Error(`Change phone is not allowed. Error: ${err}`)
+        )
+      }
+      const res = await fetch(`${USER_MANAGEMENT_URL}changeUserPhone`, {
+        method: 'POST',
+        body: JSON.stringify({ userId, phoneNumber }),
+        headers: {
+          'Content-Type': 'application/json',
+          ...authHeader
+        }
+      })
+
+      if (res.status !== 200) {
+        return await Promise.reject(
+          new Error(
+            "Something went wrong on user-mgnt service. Couldn't change user phone number"
+          )
+        )
+      }
+      return true
```

Comment:
```suggestion
```

I think we can leave the return value to just be `void` as if the promise resolves, then we know it succeeded

Reply:
Same. There were other functions in root-resolvers that returns true instead of void. so I leave it as it is for now.